telemetry/Tracks/class-tracks-event.php

```diff
@@ -0,0 +1,275 @@
+<?php
+/**
+ * Telemetry: Tracks Event class
+ *
+ * @package Automattic\VIP\Telemetry
+ */
+
+declare(strict_types=1);
+
+namespace Automattic\VIP\Telemetry;
+
+use stdClass;
+use WP_Error;
+
+/**
+ * Class that creates and validates Tracks events.
+ *
+ * @since 3.12.0
+ */
+class Tracks_Event {
+	/**
+	 * Event name prefix.
+	 */
+	protected const EVENT_NAME_PREFIX = 'viptelemetry_';
```

Comment:
We want `vip_` as the prefix.

We may also want to think about how each of our plugins could register and append their own namespace (e.g. the workflows plugin may want all events to be `vip_workflows_`)

Reply:
Yeah I've been thinking about that myself. An idea I have is to set the prefix as a constructor argument for the `Tracks` class and the `Tracks_Event` class.

Reply:
It'd be great to allow plugins to set some sort of a prefix so looking up those logs based on that would be easier

Reply:
Changed this in d9697a67ce5395ae291a52447316419ac84d2043
The prefix can now be passed to the `Tracks` class